1.Exploratory_Data_Analysis/scripts/5.anova.py

```diff
@@ -0,0 +1,385 @@
+# df_flag['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()
+# df_DMSO = df[df['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].str.contains('DMSO_0.100_DMSO_0.025')]
+# df_DMSO['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()
+# df_thapsi10 = df[df['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].str.contains('Thapsigargin_10.000_DMSO_0.025')]
+# df_thapsi10['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()
+# df_thapsi_DMSO = df[df['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].str.contains('Thapsigargin_10.000_DMSO_0.025','DMSO_0.100_DMSO_0.025')]
+# df_thapsi_DMSO['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()
+# df_thapsi1 = df[df['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].str.contains('Thapsigargin_1.000_DMSO_0.025')]
+# df_thapsi1['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()
+
+
+# ### Set up DF
+
+# In[6]:
+
+
+combined_df = pd.concat([trt1, trt2, trt3], axis=0)
+print(len(trt1), len(trt2), len(trt3), len(combined_df))
+combined_df.head(3)
+
+
+# ## Anova + Post Hoc testing (tukeyHSD)
+
+# In[7]:
+
+
+# anova for each feature in the dataframe with posthoc tukey test to determine which groups are different from each other
+lst = []
+for i in cp_features:
+    formula = f"{i} ~ C(oneb_Metadata_Treatment_Dose_Inhibitor_Dose) + C(Metadata_number_of_singlecells)"
+    model = ols(formula, combined_df).fit()
+    aov_table = sm.stats.anova_lm(model, typ=2)
+    # posthoc = MultiComparison(df_DMSO_thapsi[i], df_DMSO_thapsi['oneb_Metadata_Treatment_Dose_Inhibitor_Dose'])
+    posthoc = pairwise_tukeyhsd(
+        combined_df[i],
+        combined_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"],
+        alpha=0.001,
+    )
+    # print(posthoc)
+    lst.append([posthoc, i])
+
+
+# In[8]:
+
+
+# add all tukey test results to a dataframe from the list of tukey test results
+tukey_df = pd.DataFrame()
+for i in lst:
+    # create data frame from tukey test results with feature name as column name and the results as the rows
+    j = pd.DataFrame(i[0]._results_table.data[1:])
+    j["features"] = np.repeat(i[1], len(j))
+    tukey_df = pd.concat([tukey_df, j], axis=0)
+
+    np.repeat(i[1], len(j))
+
+tukey_df.columns = [
+    "group1",
+    "group2",
+    "meandiff",
+    "lower",
+    "upper",
+    "p-adj",
+    "reject",
+    "features",
+]
+tukey_df.head(3)
+
+
+# In[9]:
+
+
+# make new column with the absolute value of the p-adj
+tukey_df["p-adj_abs"] = abs(tukey_df["p-adj"])
+# get all p-adj values that are less than 0.05
+tukey_df_sig = tukey_df[tukey_df["p-adj_abs"] < 0.01]
+# make new column that states if the relationship is positive or negative
+tukey_df_sig["pos_neg"] = np.where(tukey_df_sig["p-adj"] > 0, "positive", "negative")
+# order the features by p-adj value
+tukey_df_sig.head(2)
+
+
+# ## Venn Diagrams
+
+# #### Venn diagram prep
+
+# In[10]:
+
+
+# get all group1 rows that are DMSO and group2 rows that are LPS 100
```

Comment:
```suggestion
# get all group1 rows that are DMSO and group2 rows that are LPS treatment
```
Based on the sh file for this script, it seems like group2 rows are any dose for LPS treatment.

Comment is the same for the below.